staging/src/k8s.io/apiserver/pkg/endpoints/handlers/watch.go

```diff
@@ -212,13 +218,21 @@ func (s *WatchServer) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	var unknown runtime.Unknown
 	internalEvent := &metav1.InternalEvent{}
 	outEvent := &metav1.WatchEvent{}
+	var bookmarkEvent *metav1.WatchEvent
```

Comment:
The bookmark concept is introduced by the cacher, I don't think it is a good idea to move the codes to watch handler.

Reply:
If we have a compelling reason we could consider it (e.g. if it would be 10x cheaper or sth like that). But I don't see any reason for now.